services/datamanager/builtin/builtin.go

```diff
@@ -517,14 +542,23 @@ func (svc *builtIn) Update(ctx context.Context, cfg *config.Config) error {
 	// Initialize or add a collector based on changes to the component configurations.
 	newCollectorMetadata := make(map[componentMethodMetadata]bool)
 	for _, attributes := range allComponentAttributes {
-		if !attributes.Disabled && attributes.CaptureFrequencyHz > 0 {
+		if !attributes.Disabled && attributes.CaptureFrequencyHz > 0 && !svc.captureDisabled {
 			componentMetadata, err := svc.initializeOrUpdateCollector(
 				attributes, updateCaptureDir)
 			if err != nil {
 				svc.logger.Errorw("failed to initialize or update collector", "error", err)
 			} else {
 				newCollectorMetadata[*componentMetadata] = true
 			}
+		} else if attributes.Disabled {
```

Comment:
Do we need this else if, or could it just be else?  Would we not want to do this if say `!attributes.Disabled && svc.captureDisabled`, for example?

Reply:
No I think, because if `svc.captureDisabled`, either no collectors existed in the past, or `toggledCaptureOff` would be true and all collectors would be closed on line 500. From there, no other collectors should have been initialized so as long as the first part of the if is true, then there wouldn't be more collectors to worry about

Reply:
Ok that makes sense, this looks good then. Definitely a bit fragile  that the logic here depends on these implicit connections over a pretty wide portion of code, which I think contributed to the bug happening. Just making a note that this is probably something we should try to simplify in the near future